### ISSUE TITLE
rxExecBy in local CC gives bad result if varsToKeep specified

### ISSUE THREAD

**Issue by hongooi73** [RevolutionAnalytics/dplyrXdf#50]
```r
mt1 <- RxXdfData("misc/mtcars.xdf", varsToKeep="cyl")
rxExecBy(mt1, "cyl", function(keys, data) head(data))

#$P1
#$P1$keys
#$P1$keys[[1]]
#$P1$keys[[1]][[1]]
#RxXdfData Source
#"d:\misc\Rtemp\Rtmpy25tZ1\tmpPartitions8a8855615e7a\data\tmpPartitions8a8855615e7a_5006475794136178589.xdf"
#fileSystem: 
    #fileSystemType: native


#$P1$keys[[2]]
#data frame with 0 columns and 1 row


#$P1$result
  #cyl
#1   6
#2   6
#3   4
#4   6
#5   8
#6   6

#$P1$status
#[1] "OK"
```

**Comment by hongooi73** [comment]
In the edge node on HDI, and with HDFS data, this happens with varsToKeep of any length:
```r
vars <- c("am", "vs", "gear")
tmpSrc <- RxXdfData("/user/sshuser/mtcarsc", fileSystem=RxHdfsFileSystem(),
    createCompositeSet=TRUE, varsToKeep=vars)
rxExecBy(tmpSrc, c("am", "gear"), function(keys, data) names(data))
#$P1
#$P1$keys
#$P1$keys[[1]]
#$P1$keys[[1]][[1]]
#RxXdfData Source
#"/tmp/RtmpYfgPc2/tmpPartitions893116ff9556/data/tmpPartitions893116ff9556_5006475794136178589.xdf"
#fileSystem:
    #fileSystemType: native


#$P1$keys[[2]]
#data frame with 0 columns and 1 row


#$P1$result
#[1] "am"   "vs"   "gear"

#$P1$status
#[1] "OK"
```

**Comment by hongooi73** [comment]
TFS item 78040

**Comment by hongooi73** [comment]
Fixed in 9.2
